Translate add-react-to-a-website (#27)
reactjs/de.reactjs.org: content/docs/add-react-to-a-website.md

---
id: add-react-to-a-website
title: React einer Webseite hinzufügen
permalink: docs/add-react-to-a-website.html
redirect_from:
  - "docs/add-react-to-an-existing-app.html"
prev: getting-started.html
next: create-a-new-react-app.html
---

Verwende so wenig oder so viel React wie du möchtest.

React wurde von Beginn an für graduelle Benutzung konzipiert und du kannst so wenig oder so viel React benutzen, wie du möchtest. Vielleicht möchtest du einer existierenden Seite nur an wenige Stellen etwas Interaktivität hinzufügen. React Komponenten sind perfekt dafür geeignet.

> Die Mehrheit der Webseiten sind und brauchen keine SPAs zu sein. Probiere React in einem kleinen Teil deiner Webseite mit **ein paar Zeilen Code und ohne build tooling** aus. Dann kannst du die Verwendung entweder graduell erweitern oder auf ein paar dynamische Widgets beschränken.

Reviewer: ```suggestion
Die Mehrheit der Webseiten sind und brauchen keine SPAs zu sein. Probiere React in einem kleinen Teil deiner Webseite mit **ein paar Zeilen Code und ohne build tooling** aus. Dann kannst du die Verwendung entweder schrittweise erweitern oder auf ein paar dynamische Widgets beschränken.
```

---

- [Füge React in einer Minute hinzu](#add-react-in-one-minute)
- [Optional: Benutze React mit JSX](#optional-try-react-with-jsx) (Kein Bundler benötigt!)

## Füge React in einer Minute hinzu {#add-react-in-one-minute}

Wir wollen in diesem Abschnitt zeigen, wie man einer existierenden HTML Seite eine React Komponente hinzufügen kann. Um zu üben kannst du deine eigene Webseite benutzen oder eine leere HTML Seite erstellen. 

Es sind keine komplizierten Werkzeuge oder Installationen nötig -- **Um diesem Abschnitt zu folgen, brauchst du nur eine Internetverbindung und eine Minute deiner Zeit.**

Optional: [Komplettes Beispiel herunterladen (2KB gezippt)](https://gist.github.com/gaearon/6668a1f6986742109c00a581ce704605/archive/f6c882b6ae18bde42dcf6fdb751aae93495a2275.zip)

### Schritt 1: Einen DOM Container dem HTML hinzufügen {#step-1-add-a-dom-container-to-the-html}

Öffne als erstes die HTML Seite, die du editieren möchtest. Füge ein leeres `<div>` Tag hinzu, um die Stelle zu markieren, an der du mit React etwas anzeigen möchtest. Zum Beispiel:

```html{3}
<!-- ... existierendes HTML ... -->

<div id="like_button_container"></div>

<!-- ... existierendes HTML ... -->
```

> Wir haben diesem `div` ein eindeutiges `id` HTML Attribut. Das erlaubt es uns später mit JavaScript wiederzufinden und darin eine React Komponente anzuzeigen.

Reviewer: ```suggestion
Wir haben diesem `div` ein eindeutiges `id` HTML Attribut gegeben. Das erlaubt es uns es später mit JavaScript wiederzufinden und darin eine React Komponente anzuzeigen.
```

>Tipp
>
>Du kannst ein "container" `<div>` wie das oben **überall** im `<body>` Tag setzen. Du kannst so viele unabhängige DOM Container in einer Seite haben wie du brauchst. Normalerweise sind sie leer -- Falls die DOM Container Inhalt besitzen, wird React diesen überschreiben.

> ### Schritt 2: Fürge die Script Tags hinzu {#step-2-add-the-script-tags}

Reviewer: ```suggestion
### Schritt 2: Füge die Script Tags hinzu {#step-2-add-the-script-tags}
```

Als nächstes fügen wir direkt vor dem schließenden `</body>` Tag drei `<script>` Tags der HTML Seite hinzu:

```html{5,6,9}
  <!-- ... anderes HTML ... -->

  <!-- React laden. -->
  <!-- Hinweis: Wenn du die Seite bereitstellen willst, ersetze "development.js" mit "production.min.js". -->
  <script src="https://unpkg.com/react@16/umd/react.development.js" crossorigin></script>
  <script src="https://unpkg.com/react-dom@16/umd/react-dom.development.js" crossorigin></script>

  <!-- Unsere React Komponente laden. -->
  <script src="like_button.js"></script>

</body>
```

Die ersten zwei Tags laden React. Der dritte läd deinen Komponenten Code.

### Schritt 3: Erstelle eine React Komponente {#step-3-create-a-react-component}

Erstelle eine Datei mit dem Namen `like_button.js` und speichere sie neben deiner HTML Seite.

Öffne **[den Starter Code](https://cdn.rawgit.com/gaearon/0b180827c190fe4fd98b4c7f570ea4a8/raw/b9157ce933c79a4559d2aa9ff3372668cce48de7/LikeButton.js)** und füge Sie in die Datei ein, die du erstellt hast.

>Tipp
>

> >Dieser Code definiert eine React Komponente mit dem Namen `LikeButton`. Mach dir keine Sorgen, falls du das noch nicht verstehst -- Wir werden die Bausteine von React später in unserem [hands-on tutorial](/tutorial/tutorial.html) und dem [main concepts guide](/docs/hello-world.html) behandeln. Jetzt wollen wir sie erstmal im Browser angezeigt bekommen!

Reviewer: ```suggestion
>Dieser Code definiert eine React Komponente mit dem Namen `LikeButton`. Mach dir keine Sorgen, falls du das noch nicht verstehst -- Wir werden die Bausteine von React später in unserem [hands-on Tutorial](/tutorial/tutorial.html) und dem [Leitfaden der Hauptkonzepte](/docs/hello-world.html) behandeln. Jetzt wollen wir sie erstmal im Browser angezeigt bekommen!
```

Nach **[dem Starter Code](https://cdn.rawgit.com/gaearon/0b180827c190fe4fd98b4c7f570ea4a8/raw/b9157ce933c79a4559d2aa9ff3372668cce48de7/LikeButton.js)**, füge zwei Zeilen am Ende von `like_button.js` an:

```js{3,4}
// ... Der Start Code, den du gerade eingefügt hast ...

const domContainer = document.querySelector('#like_button_container');
ReactDOM.render(e(LikeButton), domContainer);
```

> Diese zwei Codezeilen finden den von uns zu unserem HTML hinzugefügen `<div>` Container aus dem ersten Schritt und zeigen dann unsere React Komponente mit dem "Like" Button darin an.

Reviewer: ```suggestion
Diese zwei Codezeilen finden den von uns zu unserem HTML hinzugefügten `<div>` Container aus dem ersten Schritt und zeigen dann unsere React Komponente mit dem "Like" Button darin an.
```

### Das ist alles! {#thats-it}

Es gibt keinen vierten Schritt. **Du hast gerade deine erste React Komponente zu deiner Webseite hinzugefügt.**

Sieh dir die nächsten Abschnitte an, um mehr Tipps zu erhalten, wie du React integrieren kannst.

**[Öffne den ganzen Beispiel Source Code](https://gist.github.com/gaearon/6668a1f6986742109c00a581ce704605)**

> **[Komplette Beispiel herunterladen (2KB gezippt)](https://gist.github.com/gaearon/6668a1f6986742109c00a581ce704605/archive/f6c882b6ae18bde42dcf6fdb751aae93495a2275.zip)**

Reviewer: ```suggestion
**[Das komplette Beispiel herunterladen (2KB gezippt)](https://gist.github.com/gaearon/6668a1f6986742109c00a581ce704605/archive/f6c882b6ae18bde42dcf6fdb751aae93495a2275.zip)**
```

### Tipp: Verwende Komponeten wieder {#tip-reuse-a-component}

Häufig wirst du React Komponenten an verschiedenen Stellen einer HTML Seite anzeigen wollen. Hier ist ein Beispiel, welches den "Like" Button dreimal anzeigt und einige Daten hinzufügt.

[Öffne den ganzen Beispiel Source Code](https://gist.github.com/gaearon/faa67b76a6c47adbab04f739cba7ceda)

> [Download the full example (2KB zipped)](https://gist.github.com/gaearon/faa67b76a6c47adbab04f739cba7ceda/archive/9d0dd0ee941fea05fd1357502e5aa348abb84c12.zip)

Reviewer: ```suggestion
[Das komplette Beispiel herunterladen (2KB gezippt)](https://gist.github.com/gaearon/faa67b76a6c47adbab04f739cba7ceda/archive/9d0dd0ee941fea05fd1357502e5aa348abb84c12.zip)
```

  { onClick: () => this.setState({ liked: true }) },
  'Like'
);
```

React bietet allerdings auch die Option an, stattdessen [JSX](/docs/introducing-jsx.html) zu benutzen.

```js
// Zeige einen "Like" <button> an
return (
  <button onClick={() => this.setState({ liked: true })}>
    Like
  </button>
);
```

Diese zwei Codebeispiele sind äquivalent. Obwohl **JSX [komplett optional](/docs/react-without-jsx.html) ist**, finden es viele Leute beim Schreiben von UI Code hilfreich -- sowohl mit React als auch mit anderen Bibliotheken.

Du kannst JSX mit [diesem Online Konverter](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=Q&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2%2Cstage-3&prettier=true&targets=Node-6.12&version=6.26.0&envVersion=) ausprobieren.

### Schnell JSX ausprobieren {#quickly-try-jsx}

Du kannst JSX in deinem Projekt am schnellsten ausprobieren, indem du diesen `<script>` Tag deiner Seite hinzufügst:

```html
<script src="https://unpkg.com/babel-standalone@6/babel.min.js"></script>
```

> Jetzt kannst JSX in jedem `<script>` Tag verwenden, indem du ihm ein `type="text/babel"` Attribut hinzufügst. Hier kannst du [eine Besipiel HTML Datei mit JSX](https://raw.githubusercontent.com/reactjs/reactjs.org/master/static/html/single-file-example.html) herunterladen und mit ihr experimentieren.

Reviewer: ```suggestion
Jetzt kannst JSX in jedem `<script>` Tag verwenden, indem du diesem ein `type="text/babel"` Attribut hinzufügst. Hier kannst du [eine Besipiel HTML Datei mit JSX](https://raw.githubusercontent.com/reactjs/reactjs.org/master/static/html/single-file-example.html) herunterladen und mit ihr experimentieren.
```

Navigiere im Terminal zu deinem Projektordner und führe diese zwei Befehle aus:

1. **Schritt 1:** Führe `npm init -y` aus (falls es fehlgeschlägt, [gibt es hier Hilfe](https://gist.github.com/gaearon/246f6380610e262f8a648e3e51cad40d))
2. **Schritt 2:** Führe `npm install babel-cli@6 babel-preset-react-app@3` aus

>Tipp
>
>Wir **benutzen hier npm nur, um den JSX Präprozessor zu installieren;** Du wirst es für nichts anderes brauchen. Sowohl React als auch der Anwendungscode können ohne Änderung weiter als `<script>` Tags eingebunden werden.

Herzlichen Glückwunsch! Du hast gerade ein **JSX Setup** eingerichtet, das **für die Produktionsumgebung bereit ist**.

### Starte den JSX Präprozessor {#run-jsx-preprocessor}

Erzeuge einen Ordner mit dem Namen `src` und führe folgenden Befehl aus:

```
npx babel --watch src --out-dir . --presets react-app/prod 
```

>Hinweis
>
>`npx` ist kein Schreibfehler --  es ist ein [package runner tool, welches seit npm 5.2 mitgeliefert wird](https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b)
>

> >Falls du eine Fehlermeldung mit dem Text "You have mistakenly installed the `babel` package" siehst, hast du vielleicht den [vorherigen Schritt](#add-jsx-to-a-project) übersprungen. Führe den im gleichen Ordner aus und versuch es nochmal.

Reviewer: ```suggestion
>Falls du eine Fehlermeldung mit dem Text "You have mistakenly installed the `babel` package" siehst, hast du vielleicht den [vorherigen Schritt](#add-jsx-to-a-project) übersprungen. Führe diesen im gleichen Ordner aus und probier es nochmal.
```

Du brauchst nicht darauf zu warten, dass der Befehl fertig ausgeführt wird -- er startet einen automatisierten Watcher für JSX.

> When du jetzt eine Datei unter `src/like_button.js` mit diesem **[JSX Starter Code](https://cdn.rawgit.com/gaearon/c8e112dc74ac44aac4f673f2c39d19d1/raw/09b951c86c1bf1116af741fa4664511f2f179f0a/like_button.js)** erstellst, wird der Watcher eine präprozessierte `like_button.js` mit einfachem JavaScript Code erstellen, der im Browser ausgeführt werden kann.

Reviewer: ```suggestion
Wenn du jetzt eine Datei unter `src/like_button.js` mit diesem **[JSX Starter Code](https://cdn.rawgit.com/gaearon/c8e112dc74ac44aac4f673f2c39d19d1/raw/09b951c86c1bf1116af741fa4664511f2f179f0a/like_button.js)** erstellst, wird der Watcher eine präprozessierte `like_button.js` mit einfachem JavaScript Code erstellen, der im Browser ausgeführt werden kann.
```
